lib/msal-core/docs/token-response-types.md

```diff
@@ -0,0 +1,90 @@
+
+### Login Scopes Considerations
+
++ Within `msal@1.x.x`, `openid` and `profile` are referred to as "login scopes" or OIDC (OpenID Connect) scopes. From the library's perspective, setting the `client ID`'s value as a scope in the request configuration object is functionally equivalent to adding `openid`, `profile`, or both scopes. 
+
+    This is because during request validation, if the `client ID`'s value is present in the scopes, it is substituted with `openid` and `profile`.
+
+    Similarly, whenever `openid` or `profile` is present without the other in the scopes array, the missing login scope will be added during request validation and before the request is actually sent to the server.
+
+    This means that:
+
+    + `scopes: ['APP_CLIENT_ID_VALUE']`
+    + `scopes: ['openid']`
+    + `scopes: ['profile']`
+
+    Are all functionally equivalent to `scopes: ['openid', 'profile']` from the library's perspective. From this point on, when this document makes a reference to "login scopes", it refers to any combination or permutation of `openid`, `profile` and the value of `client ID`. The term "access token scopes", on the other hand, refers to any resource-related scope set that an access token is being requested for (i.e. 'user.read', 'mail.read', etc.). 
+
++ On a related note, `login` APIs (i.e. `loginPopup` and `loginRedirect`) will append `openid` and `profile` (without duplication) to the scopes array every time they are called, no matter what other scopes are present. This means that `login` APIs can be called with an empty, non-null scopes array in the request configuration object (as opposed to acquireToken APIs which will throw an error with an empty or null `scopes` array in the request).
+
++ If any valid access token scopes are present in the scopes array, they will remain in the array and eventually be included in the `scopes` parameter of the server request.
+
+***Important Note: `msal@1.x.x` will append `openid` and `profile` to the `scopes` parameter of every server request URL, whether or not any of these two specific scopes are passed by the developer into the API call. The presence of login scopes in the configuration object has an effect on which `response_type` value is set in the actual server request `msal1.x.x` makes, but it does not make a difference on the value of the `scopes` parameter in said server request.***
+
+The Scopes-based Response Types table below describes the mapping of `msal@1.x.x` APIs in combination with the user-provided `scopes` to the resulting `response_type` set in the server request.
+
+## Scopes-based Response Types
+
+|| Account Object in Request Configuration | Empty Scopes Array | Login Scopes Only | Access Token Scopes Only | Access Token and Login Scopes
+| -------- | ------- | ------- | -------- | --------- | -------- |
+| **loginRedirect/Popup** | **None or Matching Account in MSAL Cache** | id_token | id_token |id_token token | id_token token
```

Comment:
@pkanher617 It's been a while since we made the original table but I believe there was a reason why we thought login calls should allow access token scopes and set response type to id_token_token. Is there a need for a discussion here or should this be changed as Thomas suggests?

Reply:
To follow up: This would be a breaking change, because today you can request multiple resources in an `id_token` request in order to pre-consent to the scopes. Having this return an access token as well would result in an error if scopes spanning more than 1 resource are requested.

@pkanher617 Can correct me if I'm wrong, but I believe Auth Code is allowing `id_token token` requests on a login call. But I don't think this change should be made in core

Reply:
We have always allowed access tokens to be returned on login calls. We do allow users to request multiple resources, but generally we have restricted that through the use of extraScopesToConsent. 

@tnorling `response_type` is always `=code` in 2.0, which allows for multiple resources (doesn't matter if it is a login call or not). 

id_token token will still accept multiple resources as well, but also return an access token for the first resource (I'm not sure how this determination is made). For example, if I send `response_type=id_token token` and `scope=openid profile User.Read https://resource2/mail.read` I will get an id_token and an access token for Graph and still see a consent screen for the additional scope.

Reply:
> We have always allowed access tokens to be returned on login calls. 

I don't think this is true. Currently we set `response_type` to `id_token` on all login calls, regardless of what scopes are requested. We can _consent_ to scopes, for up to 3 resources, via a login call but it will not return an access_token. You can verify this by looking at the diff on this PR. Our messaging has been consistent that you need to login to get an id_token and then call an acquireToken method to get an access token. I don't think we should change this behavior.

I don't disagree with the statement that `id_token token` will return an access token while still allowing you to consent to other scopes, but I think we should discourage this outside of using `extraScopesToConsent` because it could lead to some confusing behaviors if someone is: A. getting only a single access token and B. Not understanding which resource you got the access token for and why

Reply:
I believe the most recent changes to the code and documentation have resolved this conversation based on the consensus reached in our last design discussion.